src/components/plots/linePlot.ts

```diff
@@ -123,6 +123,31 @@ export module Plot {
               selection: closestSelection};
     }
 
+    protected _getPlotData(xExtent: Extent, yExtent: Extent, datasetKeys: string[]): PlotData {
+      var data: any[] = [];
+      var pixelPoints: Point[] = [];
+      var selections: EventTarget[] = [];
+      datasetKeys.forEach((datasetKey) => {
+        var plotData = this.getAllPlotData(datasetKey);
+        var lineSelected = false;
+        plotData.pixelPoints.forEach((pixelPoint, index) => {
```

Comment:
[`Array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) just seems like a better way to do this. You can check if the line is selected afterwards by seeing if the filtered array is non-empty.